src/concrete/ml/quantization/quantized_ops.py

```diff
@@ -217,20 +222,113 @@ def q_impl(
         # Here we follow Eq.7 in https://arxiv.org/abs/1712.05877 to split the core computation
         # from the zero points and scales.
 
-        p = weights_q_values.shape[0]
+        p = input2_q_values.shape[-2]
+
+        # FIXME: https://github.com/zama-ai/concrete-internal/issues/512
```

Comment:
can you plase use the regular format:

```
# comment explaining the problem
# FIXME: issue id
```

Reply:
Added the issue.

Reply:
I agree with Andrei, please add a comment to explain why we have a fixme here and what we should change at one point

Reply:
Done